src/objects.js

```diff
@@ -2837,55 +2710,161 @@ SpriteMorph.prototype.blockTemplates = function (category) {
+    let button, myself = this;
+
+    function addVar(pair) {
+        var ide;
+        if (pair) {
+            if (myself.isVariableNameInUse(pair[0], pair[1])) {
+                myself.inform('that name is already in use');
+            } else {
+                ide = myself.parentThatIsA(IDE_Morph);
+                myself.addVariable(pair[0], pair[1]);
+                myself.toggleVariableWatcher(pair[0], pair[1]);
+                ide.flushBlocksCache('variables'); // b/c of inheritance
+                ide.refreshPalette();
+                ide.recordUnsavedChanges();
+            }
+        }
+    }
+
+    button = new PushButtonMorph(
+        null,
+        function () {
+            new VariableDialogMorph(
+                null,
+                addVar,
+                myself
+            ).prompt(
+                'Variable name',
+                null,
+                myself.world()
+            );
+        },
+        'Make a variable'
+    );
+    button.userMenu = this.helpMenu;
+    button.selector = 'addVariable';
+    button.showHelp = BlockMorph.prototype.showHelp;
+    return button;
+}
+
+SpriteMorph.prototype.deleteVariableButton = function () {
+    let button, myself = this;
+    button = new PushButtonMorph(
+        null,
+        function () {
+            var menu = new MenuMorph(
+                myself.deleteVariable,
+                null,
+                myself
+            );
+            myself.deletableVariableNames().forEach(name =>
+                menu.addItem(
+                    name,
+                    name,
+                    null,
+                    null,
+                    null,
+                    null,
+                    null,
+                    null,
+                    true // verbatim - don't translate
+                )
+            );
+            menu.popUpAtHand(myself.world());
+        },
+        'Delete a variable'
+    );
+    button.userMenu = this.helpMenu;
+    button.selector = 'deleteVariable';
+    button.showHelp = BlockMorph.prototype.showHelp;
+    return button;
+}
+
+SpriteMorph.prototype.devModeText = function () {
+    let txt = new TextMorph(localize('development mode \ndebugging primitives:'));
+    txt.fontSize = 9;
+    txt.setColor(this.paletteTextColor);
+    return txt;
+}
+
+SpriteMorph.prototype.helpMenu = function () {
+    // return a 1 item context menu for anything that implements a 'showHelp' method.
+    var menu = new MenuMorph(this);
+    menu.addItem('help...', 'showHelp');
+    return menu;
+}
+
+// returns an array alock templates for a selected category.
+SpriteMorph.prototype.customBlockTemplatesForCategory = function (category) {
+    var ide = this.parentThatIsA(IDE_Morph), blocks = [],
+        isInherited = false, block, inheritedBlocks;
+
+    function addCustomBlock(definition) {
+        if (definition.category === category ||
+                (Array.isArray(category) && category.includes(definition.category))) {
```

Comment:
Since I tweaked the way this worked, I decided to use the native `includes` method, rather than morphic's `contains`.

Reply:
Haha, `includes()` didn't exist when Morphic was first made (and it still doesn't exist in IE, I guess. Fine with me!